### ISSUE TITLE
weird state feature at the last timestamp of an episode

### ISSUE THREAD

**Issue by CoolPhilChen** [LARG/HFO#71]
Hi @mhauskn ! I met a quite weird phenomenon. I ran the following command to start server.
```
./bin/HFO --untouched-time=-1 --offense-agents=1 --port=6000 --frames-per-trial=50 --fullstate --trials=30 --headless
```
Then there will be 31 trials, with the first trial having 52 frames and the last invalid trial having 2 frames. If you set `-trials=n`, then there will always be n+1 trials. What's more weird, there is always a dramatic change in the last state of each episode. Take feature 53, `Ball Dist [Proximity]` as an example, and here is data of an episode with 50 states.
```
array([ 0.6400014 ,  0.6369575 ,  0.6370717 ,  0.63380563,  0.6275883 ,
        0.6319275 ,  0.6262455 ,  0.62735546,  0.6220956 ,  0.6270733 ,
        0.6219528 ,  0.62414825,  0.6172757 ,  0.6211989 ,  0.6158545 ,
        0.6068506 ,  0.6081611 ,  0.6155964 ,  0.6115577 ,  0.6169543 ,
        0.6261015 ,  0.62277246,  0.6284226 ,  0.63537216,  0.6451534 ,
        0.6534133 ,  0.66379   ,  0.671872  ,  0.66849315,  0.67407906,
        0.6691985 ,  0.6743033 ,  0.66928995,  0.6602013 ,  0.648855  ,
        0.6509881 ,  0.6425574 ,  0.63227415,  0.6179184 ,  0.6055956 ,
        0.58962023,  0.5898919 ,  0.59660244,  0.58823895,  0.57308745,
        0.57403517,  0.56837523,  0.5541458 ,  0.54245853,  0.545022  ,
       -0.20844042], dtype=float32)
```
You can see the last value must be wrong. Have you ever met such problems? Any help is well appreciated!

**Comment by mhauskn** [comment]
It's possible there could be an off-by-one error in the code that decides to stop when trials is reached. 

Sometimes I have observed that the simulator doesn't respond to the first action and sometimes it takes a frame or two before state features are initialized with correct observations. 

It seems that the last observation may be what happens when the field is reset for the next episode, rather than when the ball is in the goal.

Good observations. I would happily accept pull requests if you feel enthusiastic about fixing either problem.

**Comment by CoolPhilChen** [comment]
Yes, you are right. I just found that the last observation of each episode is just the same as the first observation of the next episode.

**Comment by mhauskn** [comment]
It may be possible to fix this by delaying the resetting of the field by
one step.
